internal/validation/config.go

```diff
@@ -0,0 +1,119 @@
+package validation
+
+import (
+	"context"
+	"fmt"
+	"github.com/speakeasy-api/openapi-generation/v2/pkg/generate"
+	sdkGenConfig "github.com/speakeasy-api/sdk-gen-config"
+	"github.com/speakeasy-api/speakeasy-client-sdk-go/v3/pkg/models/shared"
+	"github.com/speakeasy-api/speakeasy-core/events"
+	"github.com/speakeasy-api/speakeasy/internal/log"
+	"regexp"
+	"strings"
+)
+
+// ValidateConfigAndPrintErrors validates the generation config for a target and prints any errors
+// Returns an error if the config is invalid, nil otherwise
+func ValidateConfigAndPrintErrors(ctx context.Context, target string, cfg *sdkGenConfig.Config, publishingEnabled bool) error {
+	logger := log.From(ctx)
+	logger.Infof("\nValidating gen.yaml for target %s...\n", target)
+
+	logger = logger.WithFormatter(log.PrefixedFormatter)
+
+	err := events.Telemetry(ctx, shared.InteractionTypeTargetGenerate, func(ctx context.Context, event *shared.CliEvent) error {
+		errs := ValidateConfig(target, cfg, publishingEnabled)
+		if len(errs) > 0 {
+			for _, err := range errs {
+				logger.Error(fmt.Sprintf("%v", err))
+			}
+
+			return fmt.Errorf("gen.yaml config is invalid for target %s. See workflow logs for details", target)
+		}
+
+		return nil
+	})
+
+	return err
+}
+
+// ValidateConfig validates the generation config for a target and returns a list of errors
+func ValidateConfig(target string, cfg *sdkGenConfig.Config, publishingEnabled bool) []error {
+	if cfg == nil || cfg.Config == nil || cfg.Config.Languages == nil {
+		return []error{fmt.Errorf("no configuration found")}
+	} else if _, ok := cfg.Config.Languages[target]; !ok {
+		return []error{fmt.Errorf("target %s not found in configuration", target)}
+	}
+
+	return ValidateTarget(target, cfg.Config.Languages[target].Cfg, publishingEnabled)
+}
+
+func ValidateTarget(target string, config map[string]any, publishingEnabled bool) []error {
+	t, err := generate.GetTargetFromTargetString(target)
+	if err != nil {
+		return []error{err}
+	}
+
+	// TODO: newSDK???
```

Comment:
Not sure if this matters? Do we actually need to check if its a "new sdk"

Reply:
I don't think so